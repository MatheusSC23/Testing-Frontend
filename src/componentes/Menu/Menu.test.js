import { render, screen } from '@testing-library/react';
import Menu from './index';

test('Deve renderizar um link para a página inicial', () => {
  render(<Menu />);
  const linkPAginaInicial = screen.getByText('Inicial');

  expect(linkPAginaInicial).toBeInTheDocument();
});

test('Deve renderizar uma lista de links', () => {
  render(<Menu />);
  const listaDeLinks = screen.getAllByRole('link');

  expect(listaDeLinks).toHaveLength(4);
});

test('Não deve redenrizar o link para Extrato', () => {
  render(<Menu />);
  const linkExtrato = screen.queryByText('Extrato');

  expect(linkExtrato).not.toBeInTheDocument();
});

test('Deve renderizar umas lista de links com a class link', () => {
  render(<Menu />);
  const links = screen.getAllByRole('link');

  links.forEach((link) => expect(link).toHaveClass('links'));

  expect(links).toMatchSnapshot();
});
