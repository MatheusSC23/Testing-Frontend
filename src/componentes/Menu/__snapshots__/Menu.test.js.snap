// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar umas lista de links com a class link 1`] = `
Array [
  <a
    class="links"
    href="/"
  >
    Inicial
  </a>,
  <a
    class="links"
    href="/"
  >
    Transferências
  </a>,
  <a
    class="links"
    href="/"
  >
    Investimentos
  </a>,
  <a
    class="links"
    href="/"
  >
    Outros serviços
  </a>,
]
`;
